app/app.go

```diff
@@ -598,14 +618,14 @@ func New(
 
 	transferModule := ibctransfer.NewAppModule(app.TransferKeeper)
 
-	var transferStack ibcporttypes.IBCModule
-	transferStack = ibctransfer.NewIBCModule(app.TransferKeeper)
-	transferStack = ircmodule.NewIBCMiddleware(transferStack, *app.IRCKeeper, app.TransferKeeper, app.RollappKeeper, app.BankKeeper)
-	transferStack = router.NewIBCMiddleware(transferStack, app.RouterKeeper, 0, routerkeeper.DefaultForwardTransferPacketTimeoutTimestamp, routerkeeper.DefaultRefundTransferPacketTimeoutTimestamp)
+	transferIBCModule := ibctransfer.NewIBCModule(app.TransferKeeper)
+	denommetadataMiddleware := denommetadatamodule.NewIBCMiddleware(transferIBCModule, app.DenomMetadataKeeper, app.TransferKeeper, app.RollappKeeper, app.BankKeeper)
+	ircMiddleware := ircmodule.NewIBCMiddleware(denommetadataMiddleware, *app.IRCKeeper, app.TransferKeeper, app.RollappKeeper, app.BankKeeper)
+	packetForwardMiddleware := packetforwardmiddleware.NewIBCMiddleware(ircMiddleware, app.PacketForwardMiddlewareKeeper, 0, packetforwardkeeper.DefaultForwardTransferPacketTimeoutTimestamp, packetforwardkeeper.DefaultRefundTransferPacketTimeoutTimestamp)
 
 	// Create static IBC router, add transfer route, then set and seal it
 	ibcRouter := ibcporttypes.NewRouter()
-	ibcRouter.AddRoute(ibctransfertypes.ModuleName, transferStack)
+	ibcRouter.AddRoute(ibctransfertypes.ModuleName, packetForwardMiddleware)
```

Comment:
I think the `transferStack` has more meaningful name. as it's actual stack of middlewares